File: doc/codes/python/examples/api/compute/modulation/square-example.py
from autd3.units import Hz
from autd3.value import SamplingConfig
from autd3_modulation import SquareOption, modulation_buffer, square

out = modulation_buffer()

square(
    150 * Hz,
    SquareOption(
        low=0x00,
        high=0xFF,
        duty=0.5,
        sampling_config=SamplingConfig.FREQ_4K,
    ),
    out,
)

File: doc/codes/csharp/examples/api/compute/modulation/square-example.cs
using AUTD3;
using static AUTD3.Units;

// HIDE
namespace AUTD3.DocSamples.ApiComputeModulationSquareExample;

internal static class Sample
{
    internal static void Run()
    {
        // HIDE_END
var @out = Modulation.ModulationBuffer();

Modulation.Square(
    150 * Hz,
    new SquareOption(
        low: byte.MinValue,
        high: byte.MaxValue,
        duty: 0.5f,
        samplingConfig: SamplingConfig.Freq4k
    ),
    @out
);
        // HIDE
    }
}
// HIDE_END
